openshift/system/Dockerfile

```diff
@@ -1,91 +1,88 @@
-    RUBY_SCL_NAME_VERSION="${RUBY_MAJOR_VERSION}${RUBY_MINOR_VERSION}"
+# can just install memkind once it is fixed, see
+# https://issues.redhat.com/browse/RHEL-14497
+RUN cd /tmp \
+    && dnf download --source memkind \
+    && rpm2cpio memkind-*.src.rpm | cpio -idmv "memkind-*.tar.gz" \
+    && tar xvfz memkind-*.tar.gz \
+    && cd memkind-*/jemalloc/ \
+    && ./autogen.sh && ./configure --libdir=/usr/local/lib64/ && make
 
-ENV TZ=:/etc/localtime \
-    BUNDLE_GEMFILE=Gemfile \
-    VARNISH_SCL=rh-varnish6 \
-    NODEJS_SCL=rh-nodejs14 \
-    RUBY_SCL="rh-ruby${RUBY_SCL_NAME_VERSION}" \
-    GIT_SCL=rh-git227
 
-WORKDIR /opt/system
+FROM quay.io/centos/centos:stream8 AS base
 
-RUN yum-config-manager --save --setopt=cbs.centos.org_repos_sclo7-$RUBY_SCL-rh-candidate_x86_64_os_.skip_if_unavailable=true \
-    && rpm -Uvh https://download.postgresql.org/pub/repos/yum/reporpms/EL-7-x86_64/pgdg-redhat-repo-latest.noarch.rpm \
-    && yum -y update \
-    && yum remove -y postgresql \
-    && yum install -y epel-release \
-    && yum -y install centos-release-scl-rh \
-              gd-devel \
-              unixODBC-devel \
-              mysql \
-              liberation-sans-fonts \
-              llvm5.0-devel \
-              postgresql13 postgresql13-devel postgresql13-libs \
-              file \
-              $NODEJS_SCL \
-              $VARNISH_SCL-jemalloc \
-              $GIT_SCL \
-    && yum -y clean all \
-    && scl enable $VARNISH_SCL -- printf '$LIBRARY_PATH' | cut -d: -f1 > /etc/ld.so.conf.d/jemalloc.conf \
-    && ldconfig && ldconfig -p | grep jemalloc
+USER root
 
-# We don't want to redo the bundle install step every time a file has changed:
-# copying only the gemspec files and copying all the other files after the build
-ADD lib/developer_portal/*.gemspec lib/developer_portal/
-ADD vendor/active-docs/*.gemspec vendor/active-docs/
-ADD Gemfile* ./
-ADD Rakefile ./
+ENV RUBY_MAJOR_VERSION=2 \
+    RUBY_MINOR_VERSION=7 \
+    RAILS_ENV=production \
+    SAFETY_ASSURED=1 \
+    TZ=:/etc/localtime
 
-COPY openshift/system/contrib/scl_enable /opt/app-root/etc/
+ENV RUBY_VERSION="${RUBY_MAJOR_VERSION}.${RUBY_MINOR_VERSION}"
 
-ENV BASH_ENV=/opt/app-root/etc/scl_enable \
-    ENV=/opt/app-root/etc/scl_enable \
-    PROMPT_COMMAND=". /opt/app-root/etc/scl_enable" \
-    RAILS_ENV=production \
-    SAFETY_ASSURED=1
+WORKDIR /opt/system
 
-RUN source /opt/app-root/etc/scl_enable \
-    && gem install bundler --version 2.2.25 \
-    && bundle config build.pg --with-pg-config=/usr/pgsql-13/bin/pg_config \
-    && bundle config set --local without development:test:licenses \
-    && bundle install --deployment --jobs $(grep -c processor /proc/cpuinfo) --retry=5
+RUN dnf -y module enable ruby:${RUBY_VERSION} nodejs:16 mysql:8.0 \
+    && yum install -y --setopt=skip_missing_names_on_install=False,tsflags=nodocs shared-mime-info postgresql rubygem-irb rubygem-rdoc ruby libpq mysql mysql-libs gd git liberation-sans-fonts file \
```

Comment:
Better just use `dnf` instead of `yum` in all the `Dockerfile`

Reply:
I have the impression that `yum` is still the preferred way to call package manager in RHEL.

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_basic_system_settings/managing-software-packages_configuring-basic-system-settings

Reply:
I see, in that case, better use `yum` everywhere. For consistency.